### DISCUSSION TITLE
Getting debian 12 to boot from snap shots

### DISCUSSION THREAD

**Discussion by taspenwall** [SpiralLinux/SpiralLinux-project#265] [Q&A] [answered]
When playing with your OS i was able to boot from btrfs snapshots which is awesome.  As far as I know, you are the only people that have figured out how to do that on a debian or debian based system.  However spirallinux did not suit my needs as I must have my system encrypted.  I did a new install of debian using the testing reopos after coming from ubuntu.  I copied the line of thinking of making /var/lib/AccountsService and /var/lib/gdm3 as separate subvolume.  As well as /var/cache, /var/tmp  and /var/log.  My system still does not boot all the way from a snapshot and I get hung up on the Accounts daemon.   I'm wondering what other magic you did to get a system booting from ro snapshots?   Thanks

**Comment by geckolinux** [comment, marked as answer]
Hi there, I was also hitting a brick wall with that while developing SpiralLinux. I finally figured it out thanks to some of the references that I mention in the **Special thanks to:** section of the website. Specifically, this page explains which login managers need separate subvolumes:
https://www.kali.org/docs/installation/btrfs/#kali-linux-btrfs-installation-steps
You can also see which subvolumes are created for the different SpiralLinux editions (according to which login manager is used) by downloading the tarball from this Github project and inspecting: `/config/includes.chroot/etc/calamares/modules/mount.conf`

>  However spirallinux did not suit my needs as I must have my system encrypted

BTW SpiralLinux does support LUKS encryption.
